ui/src/dashboards/components/variablesControlBar/VariablesControlBar.tsx

```diff
@@ -161,13 +165,19 @@ const mstp = (state: AppState): StateProps => {
     },
   } = state
 
+  const usedVariables = filterUnusedVars(
```

Comment:
this doesn't feel like the right place to put this logic (in the view). why didn't you instead make `filterUnusedVars` better?

Reply:
I mostly didn't do that because `filterUnusedVars` is used in another place and I didn't want to create an bugs in implementing a slight modification to the function